### PR TITLE
Refactor Email delayed send implementation

### PR DESCRIPTION
**Pull request by briskt**
### Fixed
- Refactor: in `Email::send` use the return value instead of a specialized exception

_Note: this was originally included as part of PR #24. I am pulling it out as a separate PR since it is unrelated to the SMTP to SES change._

### CODE REVIEW

File: application/common/models/Email.php [hunk 1]
```diff
@@ -60,12 +60,12 @@ public function behaviors()
     }
 
     /**
-     * Attempt to send email. Returns true on success or throws exception.
+     * Attempt to send email. Returns 1 on success or 0 if not sent.
      * DOES NOT QUEUE ON FAILURE
-     * @throws \Exception if the send failed for any reason
-     * @throws EmailDelayedException if email is delayed by `send_after` or `delay_seconds`
+     * @throws \Exception if sending failed
```
Comment (forevermatt) [inline application/common/models/Email.php]: These comments seem contradictory: does it return 0 or throw an exception if the email did not send?

Comment (forevermatt) [reply application/common/models/Email.php]: Ahh, I see. The 0 is if it's not time to send it yet, and the exception is if sending was attempted but failed for some reason. That makes sense now.

File: application/common/models/Email.php [hunk 2]
```diff
@@ -60,12 +60,12 @@ public function behaviors()
     }
 
     /**
-     * Attempt to send email. Returns true on success or throws exception.
+     * Attempt to send email. Returns 1 on success or 0 if not sent.
```
Comment (forevermatt) [inline application/common/models/Email.php]:
```suggestion
     * Attempt to send email. Returns 1 on success or 0 if not attempted (such as if it's not yet time to send it).
```

File: application/common/models/Email.php [hunk 3]
```diff
@@ -98,18 +98,18 @@ public function send()
...
+     * @return int number of sent messages
      */
-    public function retry()
+    public function retry() : int
     {
         try {
-            $this->send();
-        } catch (EmailDelayedException $e) {
-            return 0;
+            return $this->send();
```
Comment (forevermatt) [inline application/common/models/Email.php:112]: Nice simplification 👍

Comment (briskt) [reply application/common/models/Email.php:112]: I think this was your suggestion a couple years ago. :)

Comment (forevermatt) [reply application/common/models/Email.php:112]: Whoops... didn't mean to pat myself on the back there.

**Review by forevermatt — APPROVED** [review]

File: application/common/models/Email.php (hunk 3 above)
Comment (briskt) [reply application/common/models/Email.php:112]: I think it's good that you're consistent!

**Pull request merged**